File: fibonacci_c++/fibonacci.py
import datetime


def fibonacci(n):
    if n <= 1:
        return n
    return fibonacci(n - 1) + fibonacci(n - 2)


if __name__ == "__main__":

    n = 50
    t1 = datetime.datetime.now()
    fib = fibonacci(n)
    t2 = datetime.datetime.now()

    time_delta = (t2 - t1).total_seconds() * 1000

    print(f"Calculated the {n}tn number of the fibonaccisequence as {fib}")
    print(f"Operation took {time_delta} miliseconds")

File: fibonacci_c++/fibonacci.cpp
#include <iostream>
#include <chrono>

float fibonacci(float n)
{
    if (n <= 1)
    {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main(int argc, char **argv)
{
    using std::chrono::duration;
    using std::chrono::duration_cast;
    using std::chrono::high_resolution_clock;
    using std::chrono::milliseconds;

    float n = 50;
    auto t1 = high_resolution_clock::now();
    float fib = fibonacci(n);
    auto t2 = high_resolution_clock::now();

    auto time_delta = duration_cast<milliseconds>(t2 - t1);

    std::cout << "Final number in fibonacci sequence is: " << fib << std::endl;
    std::cout << "Operation took" << time_delta.count() << "miliseconds." << std::endl;
}
